### PR TITLE
curvefs/client: use disk cache

### PR DESCRIPTION
**Pull request by wuhongsong**
### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?): 

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: curvefs/src/client/s3/client_s3_cache_manager.cpp [hunk 1]
```diff
@@ -318,16 +319,22 @@ int FileCacheManager::HandleReadRequest(
                 s3ClientAdaptor_->GetS3Client()->DownloadAsync(context);
             } else {
                 LOG(INFO) << "sync read s3";
-                int ret = s3ClientAdaptor_->GetS3Client()->Download(
-                    name, response.GetDataBuf() + readOffset, blockPos, n);
+                int ret;
```
Comment (ilixiaocui) [inline curvefs/src/client/s3/client_s3_cache_manager.cpp:322]: ret need initialization

File: curvefs/src/client/s3/client_s3_cache_manager.cpp [hunk 2]
```diff
@@ -1159,13 +1166,19 @@ CURVEFS_ERROR DataCache::Flush(Inode* inode, bool force) {
             }
 
             objectName = GenerateObjectName(chunkId, blockIndex);
-            int ret = s3ClientAdaptor_->GetS3Client()->Upload(
-                objectName, data_ + writeOffset, n);
+            int ret;
```
Comment (ilixiaocui) [inline curvefs/src/client/s3/client_s3_cache_manager.cpp:1169]: ditto

File: curvefs/src/client/s3/client_s3_cache_manager.cpp [hunk 3]
```diff
@@ -305,6 +305,7 @@ int FileCacheManager::HandleReadRequest(
 
             std::string name = GenerateObjectName(
                 iter->GetS3ChunkInfo().chunkid(), blockIndex);
+
             if (async) {
                 LOG(INFO) << "async read s3";
```
Comment (baijiaruo) [inline curvefs/src/client/s3/client_s3_cache_manager.cpp:310]: todo finish asnyc read， sync is a temporary solution

**Comment by wuhongsong** [comment]
recheck

**Comment by wuhongsong** [comment]
recheck

File: curvefs/src/client/s3/disk_cache_manager_impl.cpp [hunk 4]
```diff
@@ -32,20 +32,21 @@ namespace curvefs {
...
+    client_ = client;
 }
 
-int DiskCacheManagerImpl::Init(S3Client *client,
-                          const S3ClientAdaptorOption option) {
+int DiskCacheManagerImpl::Init(const S3ClientAdaptorOption option) {
     LOG(INFO) << "DiskCacheManagerImpl init start.";
     int ret;
-    ret = diskCacheManager_->Init(client, option);
+    ret = diskCacheManager_->Init(client_, option);
```
Comment (ilixiaocui) [inline curvefs/src/client/s3/disk_cache_manager_impl.cpp:43]: int ret = diskCacheManager_->Init(client_, option);

File: curvefs/src/client/s3/disk_cache_manager_impl.cpp [hunk 5]
```diff
@@ -54,6 +55,23 @@ int DiskCacheManagerImpl::Write(const std::string name,
...
+        }
+    }
+    LOG(INFO) << "write success, write name = " << name;
+    return 0;
+}
+
+int DiskCacheManagerImpl::WriteDiskFile(const std::string name,
+                          const char* buf, uint64_t length) {
+    LOG(INFO) << "write name = " << name
+              << ", length = " << length;
```
Comment (ilixiaocui) [inline curvefs/src/client/s3/disk_cache_manager_impl.cpp:74]: ret below is redundant.

**Review by baijiaruo — APPROVED** [review]

**Review by ilixiaocui — APPROVED** [review]

**Comment by wuhongsong** [comment]
recheck

**Comment by wuhongsong** [comment]
recheck

**Comment by wuhongsong** [comment]
recheck

**Comment by wuhongsong** [comment]
recheck

**Comment by wuhongsong** [comment]
recheck

**Comment by wuhongsong** [comment]
recheck

**Comment by wuhongsong** [comment]
recheck

**Pull request merged**